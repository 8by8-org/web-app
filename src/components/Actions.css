#button-style {
  color: black;
  background: #bfbfbf;
  border-radius: 0;
  border-color: #bfbfbf;
}
