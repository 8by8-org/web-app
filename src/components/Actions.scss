@import "./../styles/Components.scss";

@mixin blob-background(
  $background: url(../assets/images/Actions/TealCurve.svg)
) {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 150px;
  width: 300px;
  background-repeat: no-repeat;
  background-position: center;
  text-align: center;
  font-family: $bebas-neue;
  line-height: 77px;
  letter-spacing: 0.03em;
  background-image: $background;
}

.main-content {
  background-color: $white;
  color: $black;

  .primary-button {
    @include button;
    outline: 4px solid white;
  }

  .secondary-button {
    @include secondary-button;
  }

  .avatar-and-status {
    display: flex;
    padding: 50px;
    position: relative;
  }

  .action-status {
    width: 55%;
  }

  .avatar-container {
    @include blob-background;
    width: 100%;
    height: auto;
    #challenger-avatar {
      width: 100%;
    }
    #challenger-name {
      font-size: 30px;
    }
  }

  #action-no-challenger {
    @include heading-text($black, 36px);
    padding: 30px;
  }

  .heading {
    @include heading-text($black, 36px);
    margin: 0px 0px 0px 0px;
    letter-spacing: 0.03em;
    .underline {
      @include text-underline;
    }
  }

  .subheading {
    @include heading-text($black, 24px);
    margin: 30px 0px 15px 30px;
    line-height: 29px;
    letter-spacing: 0.03em;
  }

  .text {
    font-family: $lato;
    font-size: 18px;
    line-height: 22px;
    letter-spacing: 0.02em;
    margin: 0;
  }

  .image {
    display: block;
    margin: auto;
  }

  .action-items {
    background: $black;
    padding: 35px;
  }

  .curve {
    background-color: $black;
    width: 100%;
  }
}
