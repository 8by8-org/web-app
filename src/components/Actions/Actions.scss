@import "./../../styles/Components.scss";

@mixin blob-background(
  $background: url(./../../assets/images/Actions/TealCurve.svg)
) {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 150px;
  width: 300px;
  background-repeat: no-repeat;
  background-position: center;
  text-align: center;
  font-family: $bebas-neue;
  line-height: 77px;
  letter-spacing: 0.03em;
  background-image: $background;
}

.actions {
  background-color: $white;
  color: $black;

  .top {
    .avatar-and-status {
      display: flex;
      position: relative;
      padding: 0px 50px;
    }

    .action-status {
      width: 50%;
      font-size: 112px;
      padding-top: 10%;
    }

    .avatar-and-status-finished {
      display: flex;
      padding: 0px 50px;
      position: relative;
      flex-direction: column;
    }

    .action-status-finished {
      width: 100%;
      padding-top: 10%;
    }

    .avatar-container {
      @include blob-background;
      width: 100%;
      height: auto;
      padding: 10% 0% 10%;
      .images {
        position: relative;
      }
      #challenger-avatar {
        width: 100%;
        position: relative;
        top: 15px;
      }
      #challenger-crown {
        background-color: black;
        width: 30px;
        height: 30px;
        display: flex;
        justify-content: center;
        align-items: center;
        border-radius: 50%;
        position: absolute;
        bottom: -10px;
        right: 20px;
      }
      #challenger-name {
        @include heading-text($black, 30px);
      }
    }

    #action-no-challenger {
      @include heading-text($black, 36px);
      padding: 30px;
    }
  }

  .curve {
    background-color: $black;
    width: 100%;
  }

  .heading {
    @include heading-text($black, 48px);
    letter-spacing: 0.03em;
  }

  .subheading {
    @include subheading-text($white, 24px);
    margin: 0px 0px 30px;
    line-height: 29px;
    letter-spacing: 0.03em;
  }

  .text {
    @include text($white, 18px);
    line-height: 22px;
    letter-spacing: 0.02em;
    padding: 30px 0px 10px 0px;
    margin: 0px;
  }

  .action-items {
    background: $black;
    padding: 35px;
    color: white;
    text-align: center;

    .primary-button {
      @include button;
      // outline: 4px solid white;
    }

    .secondary-button {
      @include inverted-button;
      width: 100%;
    }
  }

  .links-container {
    display: flex;
    text-align: center;
    flex-direction: column;
  }

  .links {
    font-family: $lato;
    font-size: 18px;
    line-height: 22px;
    letter-spacing: 0.02em;
    color: $blue;
    padding-bottom: 10px;
  }

  .link-share {
    background-color: transparent;
    border: none;
    cursor: pointer;
    text-decoration: underline;
    display: inline;
    margin: 0;
    padding: 0px 0px 10px 0px;
    font-family: $lato;
    font-size: 18px;
    line-height: 22px;
    letter-spacing: 0.02em;
    color: $blue;
    font-weight: 900;
  }
}
