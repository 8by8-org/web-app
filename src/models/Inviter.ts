export type Inviter = {
  name:string;
  avatar:'1' | '2' | '3' | '4';
  inviterId:string;
}