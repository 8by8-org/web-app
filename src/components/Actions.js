import React, { useState, useEffect } from 'react'
import { Button } from "react-bootstrap";
import { useHistory } from 'react-router'
import Avatar1 from '../assets/avatars/avatar1.svg'
import Avatar2 from '../assets/avatars/avatar2.svg'
import Avatar3 from '../assets/avatars/avatar3.svg'
import Avatar4 from '../assets/avatars/avatar4.svg'
import WhiteCurve from '../assets/images/Actions/Union.svg'
import './Actions.scss'

const avatars = [Avatar1, Avatar2, Avatar3, Avatar4]

export default function Actions() {
    const history = useHistory();
    const [ loading, setLoading ] = useState(true);
    const [ challengerInfo, setChallengerInfo ] = useState(null);
    
    useEffect(() => {
        if(localStorage.getItem("challengerInfo")) {
            setChallengerInfo(JSON.parse(localStorage.getItem("challengerInfo")))
        } 
        setLoading(false)
      }, [loading]);

    return (
        loading === false?
        <>
        <div className="main-content">
            <div className="top">
                { challengerInfo ? (
                    <>
                        <div className="avatar-and-status">
                            <div className="action-status">
                                <h1>Take Action For:</h1>
                            </div>
                            <div className="avatar-container">
                                <img 
                                    src={challengerInfo.avatar ? avatars[(challengerInfo.avatar)-1] : avatars[0]} 
                                    id="challenger-avatar"
                                />
                                <p id="challenger-name">{challengerInfo.name}</p>
                            </div>
                        </div>
                    </>
                    ) : <h1 id="action-no-challenger" align="center">Take Action</h1>
                }
            </div>
            <img src={WhiteCurve} className="curve"/> 
            <div className="action-items">
                <div className="py-2">
                    <Button className="primary-button" onClick={() => {history.push(`/voterreg`); localStorage.setItem('player', 'voterreg')}}>Register to vote</Button>
                </div>
                <div className="py-2">
                    <Button className="secondary-button" onClick={() => {history.push(`/election-reminders`); localStorage.setItem('player', 'election-reminders')}}>Get election reminders</Button>
                </div>
                <div className="py-2">
                    <Button className="secondary-button" onClick={() => {history.push('/challengerwelcome'); localStorage.setItem('player', 'progress')}}>Take the challenge</Button>
                </div>
            </div>
        </div>
        </>: <h1>loading</h1>
    )
}