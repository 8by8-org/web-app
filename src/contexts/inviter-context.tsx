import { Inviter } from "@/models/Inviter";
import { createContext, PropsWithChildren, useState } from "react";

export type InviterContextType = {
  inviterInfo:InviterInfo;
}

type InviterInfo = Inviter | null;

//setting the default context to {activeUser:null} allows for testing unwrapped components
export const InviterContext = createContext<InviterContextType>({inviterInfo: null});

export function InviterContextProvider({children}:PropsWithChildren) {
  const [inviterInfo,setInviterInfo] = useState<InviterInfo>(null);
  return <InviterContext.Provider value={{inviterInfo}}>{children}</InviterContext.Provider>
}

