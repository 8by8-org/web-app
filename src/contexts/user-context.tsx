import {createContext, PropsWithChildren, useState } from "react";
import { User } from "@/models/User";
import { UserType } from "@/models/UserType";

export type UserContextType = {
  activeUser:ActiveUser;
}

type ActiveUser = User | null;

//setting the default context to {activeUser:null} allows for testing unwrapped components
export const UserContext = createContext<UserContextType>({activeUser: null});

export function UserContextProvider({children}:PropsWithChildren) {
  const [activeUser,setActiveUser] = useState<ActiveUser>(null);
  return <UserContext.Provider value={{activeUser}}>{children}</UserContext.Provider>
}

