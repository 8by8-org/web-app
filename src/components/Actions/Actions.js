import React, { useState, useEffect } from "react";
import { getUserDatabase } from "./../../functions/UserData";
import { Button } from "react-bootstrap";
import { useHistory } from "react-router";
import { useAuth } from "../../contexts/AuthContext";
import { getFirestore, getDoc, doc } from "firebase/firestore";
import Avatar1 from "./../../assets/avatars/avatar1.svg";
import Avatar2 from "./../../assets/avatars/avatar2.svg";
import Avatar3 from "./../../assets/avatars/avatar3.svg";
import Avatar4 from "./../../assets/avatars/avatar4.svg";
import WhiteCurve from "./../../assets/images/Actions/Union.svg";
import Crown from "./../../assets/images/Actions/Crown.svg";
import ConfettiAnimation from "./../ConfettiAnimation";
import Invite from "../Invite.js";
import "./Actions.scss";
import { LoadingWheel } from "../LoadingWheel/LoadingWheel.component";

const avatars = [Avatar1, Avatar2, Avatar3, Avatar4];

export default function Actions() {
  const history = useHistory();
  const { currentUser } = useAuth();
  const [loading, setLoading] = useState(false);
  const [challengerInfo, setChallengerInfo] = useState(null);
  const [registeredVoter, setRegisteredVoter] = useState(false);
  const [notifyElectionReminders, setNotifyElectionReminders] = useState(false);
  const [startedChallenge, setStartedChallenge] = useState(false);
  const toggleInvite = React.useRef();

  useEffect(() => {
    if (localStorage.getItem("challengerInfo")) {
      setChallengerInfo(JSON.parse(localStorage.getItem("challengerInfo")));
      if (currentUser) {
        fetchUserData();
      } else {
        setLoading(true);
      }
      setLoading(false);
    } else {
      if (currentUser) {
        getUserDatabase()
          .then((data) => {
            if (data.invitedBy) {
              setRegisteredVoter(data.isRegisteredVoter);
              setStartedChallenge(data.startedChallenge);
              setNotifyElectionReminders(data.notifyElectionReminders);
              getChallengerInfo(data.invitedBy);
              setLoading(false);
            } else {
              history.push(`/signin`);
            }
          })
          .catch((e) => console.log(e));
      } else {
        history.push(`/signin`);
      }
    }
  }, []);

  function fetchUserData() {
    getUserDatabase()
      .then((data) => {
        setRegisteredVoter(data.isRegisteredVoter);
        setStartedChallenge(data.startedChallenge);
        setNotifyElectionReminders(data.notifyElectionReminders);
        setLoading(true);
      })
      .catch((e) => console.log(e));
  }

  async function getChallengerInfo(invitedBy) {
    const db = getFirestore();
    const docRef = doc(db, "users", invitedBy);
    const query = await getDoc(docRef);
    const info = (({ name, avatar }) => ({ name, avatar }))(query.data());
    info.challengerID = invitedBy;
    setChallengerInfo(JSON.stringify(info));
  }

  return loading === false ? (
    <div>
      {/* if all three actions are completed */}
      {registeredVoter && notifyElectionReminders && startedChallenge ? (
        <div className="actions">
          <ConfettiAnimation time={8000} />

          <div className="top">
            <div className="avatar-and-status-finished">
              <div className="action-status-finished">
                <h1 className="heading">
                  You are done!
                  <br />
                  You Supported:
                </h1>
              </div>

              <div className="avatar-container">
                <div className="images">
                  <img
                    alt="Challenger Avatar"
                    src={
                      challengerInfo.avatar
                        ? avatars[challengerInfo.avatar - 1]
                        : avatars[0]
                    }
                    id="challenger-avatar"
                  />
                </div>

                <p id="challenger-name">{challengerInfo.name}</p>
              </div>
            </div>
          </div>

          <img alt="White Curve" src={WhiteCurve} className="curve" />

          <div className="action-items">
            <div className="py-2">
              <Button
                className="primary-button"
                onClick={() => {
                  history.push(`/signin`);
                }}
              >
                See Your Challenge
              </Button>
            </div>

            <div className="py-2">
              <Button
                className="secondary-button"
                onClick={() => toggleInvite.current()}
              >
                Share About Your Actions
              </Button>
            </div>
          </div>
        </div>
      ) : (
        // when there are actions that can still be completed by the user
        <div className="actions">
          <div className="top">
            <div className="avatar-and-status">
              <div className="action-status">
                {registeredVoter ||
                notifyElectionReminders ||
                startedChallenge ? (
                  <h1 className="heading">You Took Action!</h1>
                ) : (
                  <h1 className="heading">Take Action For:</h1>
                )}
              </div>

              <div className="avatar-container">
                <div className="images">
                  <img
                    alt="Challenger Avatar"
                    src={
                      challengerInfo.avatar
                        ? avatars[challengerInfo.avatar - 1]
                        : avatars[0]
                    }
                    id="challenger-avatar"
                  />
                  {(registeredVoter ||
                    notifyElectionReminders ||
                    startedChallenge) && (
                    <div id="challenger-crown">
                      <img
                        alt="Challenger Crown"
                        src={Crown}
                        className="crown"
                      />
                    </div>
                  )}
                </div>

                <p id="challenger-name">{challengerInfo.name}</p>
              </div>
            </div>
          </div>

          <img alt="White Curve" src={WhiteCurve} className="curve" />

          <div className="action-items">
            {/* when only Take the challenge action is left */}
            {registeredVoter && notifyElectionReminders && !startedChallenge ? (
              <h6 className="subheading">
                OTHER ACTIONS YOU CAN TAKE TO
                <br />
                HELP THE AAPI COMMUNITY
              </h6>
            ) : (
              // when there is at least one action completed
              (registeredVoter ||
                notifyElectionReminders ||
                startedChallenge) && (
                <h6 className="subheading">
                  {challengerInfo.name} GOT A BADGE!
                  <br />
                  HERE ARE OTHER ACTIONS TO HELP
                  <br />
                  THE AAPI COMMUNITY
                  <br />
                </h6>
              )
            )}

            {/* action buttons only displayed when they are not completed */}
            {!registeredVoter && (
              <div className="py-2">
                <Button
                  className="primary-button"
                  onClick={() => {
                    history.push(`/voterreg`);
                    localStorage.setItem("player", "voterreg");
                  }}
                >
                  Register to vote
                </Button>
              </div>
            )}

            {!notifyElectionReminders && (
              <div className="py-2">
                <Button
                  className={
                    registeredVoter ? "primary-button" : "secondary-button"
                  }
                  onClick={() => {
                    history.push(`/election-reminders`);
                    localStorage.setItem("player", "election-reminders");
                  }}
                >
                  Get election reminders
                </Button>
              </div>
            )}

            {!startedChallenge && (
              <div className="py-2">
                <Button
                  className={
                    registeredVoter && notifyElectionReminders
                      ? "primary-button"
                      : "secondary-button"
                  }
                  onClick={() => {
                    history.push("/challengerwelcome");
                    localStorage.setItem("player", "progress");
                  }}
                >
                  Take the challenge
                </Button>
              </div>
            )}

            {registeredVoter && notifyElectionReminders && !startedChallenge ? (
              // when only Take the challenge action is left
              <div>
                <h6 className="text">
                  Voting matters towards better AAPI
                  <br />
                  representation in our country. You can help
                  <br />
                  the community by encouring others to
                  <br />
                  register! Take the challenge on your own!
                  <br />
                </h6>
                <div className="links-container">
                  <a href="/homepage" className="links">
                    What is 8by8 Challenge?
                  </a>
                  <button
                    type="button"
                    className="link-share"
                    onClick={() => toggleInvite.current()}
                  >
                    Share about your actions
                  </button>
                </div>
              </div>
            ) : (
              // when at least one action is completed
              <div>
                {/* these are for when only that specific action is completed */}
                {registeredVoter &&
                  !notifyElectionReminders &&
                  !startedChallenge && (
                    <h6 className="text">Thanks for registering to vote!</h6>
                  )}

                {!registeredVoter &&
                  notifyElectionReminders &&
                  !startedChallenge && (
                    <h6 className="text">
                      Thanks for getting election reminders!
                    </h6>
                  )}

                {!registeredVoter &&
                  !notifyElectionReminders &&
                  startedChallenge && (
                    <h6 className="text">Thanks for taking the challenge!</h6>
                  )}

                {/* this is for when registered to vote or election reminders are turned on and challenge has been started */}
                {(registeredVoter || notifyElectionReminders) &&
                  startedChallenge && (
                    <h6 className="text">Thanks for your actions!</h6>
                  )}

                <div className="links-container">
                  {/* this is for when registered to vote or election reminders are turned on */}
                  {((registeredVoter &&
                    !notifyElectionReminders &&
                    !startedChallenge) ||
                    (!registeredVoter &&
                      notifyElectionReminders &&
                      !startedChallenge)) && (
                    <button
                      type="button"
                      className="link-share"
                      onClick={() => toggleInvite.current()}
                    >
                      Share about your action
                    </button>
                  )}

                  {/* this is for when registered to vote or election reminders are turned on and challenge has been started */}
                  {(registeredVoter || notifyElectionReminders) &&
                    startedChallenge && (
                      <button
                        type="button"
                        className="link-share"
                        onClick={() => toggleInvite.current()}
                      >
                        Share about your actions
                      </button>
                    )}

                  {/* this is for when user has started their own challenge */}
                  {startedChallenge && (
                    <a href="/signin" className="links">
                      See your challenge
                    </a>
                  )}
                </div>
              </div>
            )}
          </div>
        </div>
      )}

      <Invite toggleInvite={toggleInvite} isShare={true} />
    </div>
  ) : (
    <LoadingWheel overlay={false} />
  );
}
