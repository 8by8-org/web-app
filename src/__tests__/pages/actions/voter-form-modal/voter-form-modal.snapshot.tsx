import { InviterContext, InviterContextType } from '@/contexts/inviter-context';
import { UserContext, UserContextType } from '@/contexts/user-context';
import { Inviter } from '@/models/Inviter';
import { User } from '@/models/User';
import Actions from '@/pages/actions';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Builder } from 'builder-pattern';

import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

describe('Actions', () => {
  afterEach(cleanup);
    
  it('checks if the voter form modal appears when the user has registered to vote and clicks on the \"Get your registration form again\" button', async () => {
    const userInfo = Builder<User>().name('Test').avatar('1').registeredVoter(true).notifyElectionReminders(false).startedChallenge(false).build();
    const userCtxValue = Builder<UserContextType>().activeUser(userInfo).build();
    const inviter = Builder<Inviter>().name('Test Name').avatar(1).finishedChallenge(true).build();
    const inviterCtxValue = Builder<InviterContextType>().inviterInfo(inviter).build();

    const user = userEvent.setup();

    HTMLDialogElement.prototype.show = jest.fn(function mock(
      this: HTMLDialogElement
    ) {
      this.open = true;
    });
    
    HTMLDialogElement.prototype.showModal = jest.fn(function mock(
      this: HTMLDialogElement
    ) {
      this.open = true;
    });
    
    HTMLDialogElement.prototype.close = jest.fn(function mock(
      this: HTMLDialogElement
    ) {
      this.open = false;
    });

    const { container } = render(
      <UserContext.Provider value={userCtxValue}>
        <InviterContext.Provider value={inviterCtxValue}>
          <Actions />
        </InviterContext.Provider>
      </UserContext.Provider>
    );
    
    const voterFormModalLink = await screen.findByText('Get your registration form again');
    await user.click(voterFormModalLink); 
    expect(container).toMatchSnapshot();
  });
});