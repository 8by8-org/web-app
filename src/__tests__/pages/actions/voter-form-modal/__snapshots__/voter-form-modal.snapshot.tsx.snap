// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions checks if the voter form modal appears when the user has registered to vote and clicks on the "Get your registration form again" button 1`] = `
<div>
  <div
    class="outer_container"
  >
    <div
      class="inner_container"
    >
      <div>
        <section
          class="section_1"
        >
          <div
            class="status"
          >
            <h1
              class="heading"
            >
              <span>
                Challenge 
                <br />
                Won! 
              </span>
            </h1>
            <div
              class="avatar_container"
            >
              <div>
                <img
                  alt="Inviter Avatar"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  id="inviter-avatar"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <img
                  alt="Inviter Crown"
                  class="crown"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
              <h3
                class="inviter_name"
                id="inviter-name"
              >
                Test Name
              </h3>
            </div>
          </div>
        </section>
        <div
          class="curve_container"
        >
          <img
            alt="White Curve"
            class="white_curve"
            data-nimg="1"
            decoding="async"
            draggable="false"
            height="40"
            id="white_curve"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <section
          class="section_2"
        >
          <div>
            <h3
              class="subheading"
            >
              <span>
                Test Name
                 won the 8by8 challenge, 
                <br />
                 and you get a reward!
              </span>
            </h3>
            <button
              class="first_button"
              type="button"
            >
              <span>
                Get election reminders
              </span>
            </button>
            <button
              class="second_button"
              type="button"
            >
              <span>
                Take the challenge
              </span>
            </button>
            <div>
              <h6
                class="text"
              >
                Thanks for registering to vote!
              </h6>
              <div
                class="links_container"
              >
                <div>
                  <button
                    class="button_link"
                    type="button"
                  >
                    Get your registration form again
                  </button>
                </div>
                <button
                  class="button_link"
                  type="button"
                >
                  Share about your action
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
      <dialog
        aria-label="Something went wrong."
        class="popup_modal_light_theme"
        open=""
        role="dialog"
      >
        <div>
          <div
            class="close_btn_container"
          >
            <button
              aria-label="close dialog"
              class="close_btn"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </button>
          </div>
          <div
            class="content"
          >
            <p
              class="modal_heading"
            >
              We emailed you!
            </p>
            <p
              class="modal_text"
            >
              Check your email to get your voter registration PDF form.
            </p>
            <button
              class="modal_ok_btn"
              type="button"
            >
              OK
            </button>
          </div>
        </div>
      </dialog>
    </div>
  </div>
</div>
`;
